### ISSUE TITLE
Transcode unspupported files automatically 

### ISSUE THREAD

**Issue by muhlinux** [UnicornsOnLSD/finamp#91]
When playing unsupported (e.g., opus on iOS) files, transcode them automatically instead of failing.

Log from tying to play an opus file:
`[AudioServiceHelper/SEVERE] 2021-07-29 11:28:23.398423: PlatformException((-11828) Cannot Open, null, null, null)`

**Comment by KNIX3** [comment]
I would also like to ask why everyone keeps saying opus on iOS is not natively supported. According to [this article](https://www.xda-developers.com/ios-11-android-support-opus-audio-codec-mp3-successor/), iOS 11 added opus support and that was in 2017. What am I missing?

**Comment by UnicornsOnLSD** [comment]
It sort of did, but only when in the [Core Audio Format](https://en.wikipedia.org/wiki/Core_Audio_Format). This makes it pretty useless, since OGG is the container that everyone actually uses.

**Comment by KNIX3** [comment]
Oh. That's upsetting.

**Comment by UnicornsOnLSD** [comment]
Yeah it's really annoying. I'm thinking of making a GStreamer backend so that I don't have to deal with iOS's crappy player stuff, but that's a long way off

**Comment by Chaphasilor** [comment]
Comment originally by @felix920506 on Discord:
> Maybe some form of checkbox system where codecs with support problems can be checked/unchecked by the user?
Seems like a nice way to do this, where we can have a default selection, but users can customize this if needed (e.g. older device with less support, or new device/update with better support)
